status/private/protocol/chat2_message.nim

```diff
@@ -0,0 +1,45 @@
+import # std libs
+  std/times
+
+import # vendor libs
+  libp2p/protobuf/minprotobuf
+
+export ProtoBuffer, ProtoResult
+
+type
+  Chat2Message* = object
+    nick*: string
+    payload*: seq[byte]
+    timestamp*: int64
+
+  Chat2MessageResult* = ProtoResult[Chat2Message]
```

Comment:
```suggestion
  Chat2Error* = enum
    GetFieldError = "chat2: error getting field from protobuffer"

  Chat2MessageResult*[T] = ProtoResult[T]
  
proc mapErrTo[T, E](r: Result[T, E], v: static Chat2Error): ProtoResult[T] =
  {.raises: [].} =
	
  r.mapErr(proc (e: E): Chat2Error = v)
```

Reply:
This one may be a bit off, i.e. `Chat2MessageResult*` would never have cause to be generic; rather it's a specific `ProtoResult[T]` where `T` is always `Chat2Message`.

Reply:
My line of thinking is that we are adding a new layer on top of the existing `ProtoResult[T]` layer. The errors we present to layers above likely should be a bit different to he lower-level waku errors.

Reply:
I get the part about the error, but I don't understand how that relates to:
```nim
Chat2MessageResult*[T] = ProtoResult[T]
```
being preferable to:
```nim
type Chat2MessageResult* = ProtoResult[Chat2Message]
```

Maybe the `mapErrorTo` approach won't work with the latter. I'll experiment, want to understand this apsect better.